### ISSUE TITLE
Incompatibility with springboot 3.4 - java.lang.NoClassDefFoundError: org/springframework/util/Base64Utils

### ISSUE THREAD

**Issue by ahoehma** [codecentric/spring-boot-admin#3863]
## Spring Boot Admin Server information

- **Version**:
3.3.6

- **Spring Boot version**:
3.4.0

```
java.lang.NoClassDefFoundError: org/springframework/util/Base64Utils
	at de.codecentric.boot.admin.server.web.client.BasicAuthHttpHeaderProvider.encode(BasicAuthHttpHeaderProvider.java:89)
	at de.codecentric.boot.admin.server.web.client.BasicAuthHttpHeaderProvider.getHeaders(BasicAuthHttpHeaderProvider.java:83)
	at de.codecentric.boot.admin.server.web.client.CompositeHttpHeadersProvider.lambda$getHeaders$0(CompositeHttpHeadersProvider.java:36)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1597)
	at de.codecentric.boot.admin.server.web.client.CompositeHttpHeadersProvider.getHeaders(CompositeHttpHeadersProvider.java:36)
	at de.codecentric.boot.admin.server.web.client.InstanceExchangeFilterFunctions.lambda$addHeaders$0(InstanceExchangeFilterFunctions.java:70)
	at org.springframework.web.reactive.function.client.DefaultClientRequestBuilder.headers(DefaultClientRequestBuilder.java:114)
	at de.codecentric.boot.admin.server.web.client.InstanceExchangeFilterFunctions.lambda$addHeaders$1(InstanceExchangeFilterFunctions.java:70)
	at de.codecentric.boot.admin.server.web.client.InstanceWebClient.lambda$toExchangeFilterFunction$4(InstanceWebClient.java:68)
	at java.base/java.util.Optional.map(Optional.java:260)
	at de.codecentric.boot.admin.server.web.client.InstanceWebClient.lambda$toExchangeFilterFunction$6(InstanceWebClient.java:68)
	at org.springframework.web.reactive.function.client.ExchangeFilterFunction.lambda$andThen$0(ExchangeFilterFunction.java:62)
	at reactor.core.publisher.MonoFlatMap$FlatMapMain.onNext(MonoFlatMap.java:132)
	at reactor.core.publisher.FluxSwitchIfEmpty$SwitchIfEmptySubscriber.onNext(FluxSwitchIfEmpty.java:74)
	at reactor.core.publisher.FluxMapFuseable$MapFuseableSubscriber.onNext(FluxMapFuseable.java:129)
	at reactor.core.publisher.Operators$ScalarSubscription.request(Operators.java:2571)
	at reactor.core.publisher.FluxMapFuseable$MapFuseableSubscriber.request(FluxMapFuseable.java:171)
	at reactor.core.publisher.Operators$MultiSubscriptionSubscriber.set(Operators.java:2367)
	at reactor.core.publisher.Operators$MultiSubscriptionSubscriber.onSubscribe(Operators.java:2241)
	at reactor.core.publisher.FluxMapFuseable$MapFuseableSubscriber.onSubscribe(FluxMapFuseable.java:96)
	at reactor.core.publisher.MonoJust.subscribe(MonoJust.java:55)
	at reactor.core.publisher.InternalMonoOperator.subscribe(InternalMonoOperator.java:76)
	at reactor.core.publisher.MonoDefer.subscribe(MonoDefer.java:53)
	at reactor.core.publisher.InternalMonoOperator.subscribe(InternalMonoOperator.java:76)
	at reactor.core.publisher.MonoDeferContextual.subscribe(MonoDeferContextual.java:55)
	at reactor.core.publisher.InternalMonoOperator.subscribe(InternalMonoOperator.java:76)
	at reactor.core.publisher.MonoFlatMap$FlatMapMain.onNext(MonoFlatMap.java:165)
	at reactor.core.publisher.Operators$ScalarSubscription.request(Operators.java:2571)
	at reactor.core.publisher.MonoFlatMap$FlatMapMain.request(MonoFlatMap.java:194)
	at reactor.core.publisher.MonoFlatMap$FlatMapMain.request(MonoFlatMap.java:194)
	at reactor.core.publisher.Operators$MultiSubscriptionSubscriber.set(Operators.java:2367)
	at reactor.core.publisher.Operators$MultiSubscriptionSubscriber.onSubscribe(Operators.java:2241)
	at reactor.core.publisher.MonoFlatMap$FlatMapMain.onSubscribe(MonoFlatMap.java:117)
	at reactor.core.publisher.MonoFlatMap$FlatMapMain.onSubscribe(MonoFlatMap.java:117)
	at reactor.core.publisher.MonoJust.subscribe(MonoJust.java:55)
	at reactor.core.publisher.MonoDefer.subscribe(MonoDefer.java:53)
	at reactor.core.publisher.InternalMonoOperator.subscribe(InternalMonoOperator.java:76)
	at reactor.core.publisher.FluxRetryWhen.subscribe(FluxRetryWhen.java:81)
	at reactor.core.publisher.MonoRetryWhen.subscribeOrReturn(MonoRetryWhen.java:46)
	at reactor.core.publisher.Mono.subscribe(Mono.java:4560)
	at reactor.core.publisher.FluxFlatMap$FlatMapMain.onNext(FluxFlatMap.java:430)
	at reactor.core.publisher.FluxMapFuseable$MapFuseableSubscriber.onNext(FluxMapFuseable.java:129)
	at reactor.core.publisher.FluxFilterFuseable$FilterFuseableSubscriber.tryOnNext(FluxFilterFuseable.java:148)
	at reactor.core.publisher.FluxIterable$IterableSubscriptionConditional.slowPath(FluxIterable.java:664)
	at reactor.core.publisher.FluxIterable$IterableSubscriptionConditional.request(FluxIterable.java:623)
	at reactor.core.publisher.FluxFilterFuseable$FilterFuseableSubscriber.request(FluxFilterFuseable.java:191)
	at reactor.core.publisher.FluxMapFuseable$MapFuseableSubscriber.request(FluxMapFuseable.java:171)
	at reactor.core.publisher.FluxFlatMap$FlatMapMain.onSubscribe(FluxFlatMap.java:373)
	at reactor.core.publisher.FluxMapFuseable$MapFuseableSubscriber.onSubscribe(FluxMapFuseable.java:96)
	at reactor.core.publisher.FluxFilterFuseable$FilterFuseableSubscriber.onSubscribe(FluxFilterFuseable.java:87)
	at reactor.core.publisher.FluxIterable.subscribe(FluxIterable.java:179)
	at reactor.core.publisher.FluxIterable.subscribe(FluxIterable.java:83)
	at reactor.core.publisher.Mono.subscribe(Mono.java:4576)
	at reactor.core.publisher.FluxConcatMapNoPrefetch$FluxConcatMapNoPrefetchSubscriber.onNext(FluxConcatMapNoPrefetch.java:207)
	at reactor.core.publisher.FluxSubscribeOn$SubscribeOnSubscriber.onNext(FluxSubscribeOn.java:151)
	at reactor.core.publisher.FluxPeek$PeekSubscriber.onNext(FluxPeek.java:200)
	at reactor.core.publisher.FluxPeek$PeekSubscriber.onNext(FluxPeek.java:200)
	at reactor.core.publisher.FluxInterval$IntervalRunnable.run(FluxInterval.java:124)
	at reactor.core.scheduler.PeriodicWorkerTask.call(PeriodicWorkerTask.java:59)
	at reactor.core.scheduler.PeriodicWorkerTask.run(PeriodicWorkerTask.java:73)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:572)
	at java.base/java.util.concurrent.FutureTask.runAndReset(FutureTask.java:358)
	at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:305)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	at java.base/java.lang.Thread.run(Thread.java:1575)
```

Related to MR https://github.com/codecentric/spring-boot-admin/pull/3860

**Comment by SteKoe** [comment]
Hi @ahoehma,

as you can see, we are currently working to support Spring Boot 3.4. Right now SBA is not working, as we use deprecated features that have been removed in Spring Boot 3.4. - Until the release of SBA 3.4.0, you have to use latest 3.3 release.

**Comment by erikpetzold** [comment]
this has already been fixed, but not yet released: https://github.com/codecentric/spring-boot-admin/pull/3858

**Comment by ahoehma** [comment]
As temp. workaround/hack I put in my sba-server-application this file :)

```
package org.springframework.util;

import java.util.Base64;

import lombok.AccessLevel;
import lombok.NoArgsConstructor;

@Deprecated
@NoArgsConstructor(access = AccessLevel.PRIVATE)
public final class Base64Utils {

  public static String encodeToString(final byte[] bytes) {
    return Base64.getEncoder().encodeToString(bytes);
  }

}
```

**Comment by SteKoe** [comment]
Thanks for sharing a workaround! We have just released version 3.4.0 to maven central! Hope this fixes the Base64Utils issue for you, too!
